### ISSUE TITLE
"0x89 is an invalid start of a value" on Canvas App Unpack, pac v1.19.4

### ISSUE THREAD

**Issue by concoursdude** [microsoft/powerplatform-build-tools#221]
Previous issue with unpacking (NullRef exception) seems to be resolved. Receiving a new error now on unpack as shown below. PP-BT v2.0.8
![image](https://user-images.githubusercontent.com/51216063/197239873-3f5ff12a-e718-494d-8a2a-e69188358b2b.png)

**Comment by davidjenni** [comment]
@concoursdude  is that a repro with the same .msapp that you already had a conversation with @tomjschneider that resulted in the fix we shipped as part of 1.19.4? Tom will likely need that .msapp for a repro/investigation.
Don't attach your .msapp here, one option is to email it via: <mailto:PPTools@microsoft.com>

**Comment by tomjschneider** [comment]
Sorry to hear you're still experiencing an issue here. Feel free to send any .msapps directly to me as well and I'll get them to the team for analysis.

**Comment by davidjenni** [comment]
@concoursdude please paste a bit more of the PP-BT task's AzDO log, i.e. the part where the task calls into pac CLI.

Paste just enough from that log (feel free to redact some of the info), or send a more complete AzDO log to the above email address. Thx

**Comment by concoursdude** [comment]
On it @davidjenni and @tomjschneider -- you can breathe a sigh of relief that this is, in fact, a different msapp file throwing this error. Sorry that I didn't mention that in the OP. 

I will send the msapp to the above address next week. In the interim, here's more of the log: 

![image](https://user-images.githubusercontent.com/51216063/197291721-e7b7e054-04a9-4cff-92a3-0444591eb5e1.png)

**Comment by concoursdude** [comment]
@davidjenni @tomjschneider Just sent the .msapp to the address posted above. Let me know if you need anything else.

**Comment by davidjenni** [comment]
@concoursdude thanks, I've received your .msapp and forwarded it to @tomjschneider's corp email

**Comment by tomjschneider** [comment]
@concoursdude Looks like this is a different issue in the unpacking code. My apologies that you're hitting another error with this. I will go ahead and get a bug filed for this and get it in front of the team to resolve. Like last time, I will let you know when it is resolved, and when it has been merged into the PAC CLI code.

If you need me to investigate a workaround as well, let me know and I can do so.

**Comment by concoursdude** [comment]
Not a problem @tomjschneider , glad we can be of assistance in ironing these out. I'm actually waiting on an enhancement that's currently in the December backlog before we go live with PP-ALM for this particular client, anyway, so this is as good of a time as any to be running into issues. Let me know if there's anything else I can provide to you to help. Thanks.

**Comment by tomjschneider** [comment]
@concoursdude Apologies for the delay in getting back to you on this. We did resolve this issue late last year, and it should have gone out with one of the early PAC CLI releases this year. I just tested the latest version, and it appears to be fixed. Let me know if you're still seeing this issue or any other issues.

**Comment by concoursdude** [comment]
@tomjschneider Thanks for the update, but the collision problem seems to be persistent in the newest releases. Here's a pipeline run I attempted this morning: 

![image](https://user-images.githubusercontent.com/51216063/226978835-a0ae5c8e-b38d-4156-85b3-cd1ca76e6f86.png)

**Comment by tomjschneider** [comment]
@concoursdude That is strange... is this the same app you sent me last time or a different app? Would you mind sending me the latest .msapp you have which is having this issue?

**Comment by concoursdude** [comment]
@tomjschneider Yep, same app. I can send you the .msapp.

**Comment by tomjschneider** [comment]
@concoursdude Looking again at your screenshot, it looks like your pipeline is running against PAC CLI version `1.21.5`, which contains the `0.5.7-preview` version of our library. This issue was fixed in the next version of our library, which explains why you're still seeing this. Could you update to the latest version of the PAC CLI, `1.21.13`, and confirm if you're still experiencing this issue?

**Comment by concoursdude** [comment]
> @concoursdude Looking again at your screenshot, it looks like your pipeline is running against PAC CLI version `1.21.5`, which contains the `0.5.7-preview` version of our library. This issue was fixed in the next version of our library, which explains why you're still seeing this. Could you update to the latest version of the PAC CLI, `1.21.13`, and confirm if you're still experiencing this issue?

Does that update not happen automatically? Do I need to sync my pipelines? I've never had to manually update the PAC CLI version that I can recall..

**Comment by tomjschneider** [comment]
Good question - I'm actually not terribly familiar with the Power Platform build tools, just the specifics of our Canvas pack/unpack. @tehcrashxor Would you know more about how to check this or how to ensure that the latest version of the PAC CLI is being using in the pipeline?

**Comment by tehcrashxor** [comment]
@tomjschneider, PPBT and the GH Actions do not use the _latest_ PAC CLI -- they use an explicit version of PAC CLI hard coded for each version of PPBT and GH Actions.  This is due to having no programably enforceable contract between the command line args  of the C# based PAC CLI and the Task args sent to it from the NodeJS based ADO / GH Actions, so we do not update the version of PAC CLI used unless we know of bug fixes or new features that we need to ship for the pipeline tools.

We can look at updating to a new PAC version here in the next release cycle (probably next week?) to get your fix in.

**Comment by tomjschneider** [comment]
> @tomjschneider, PPBT and the GH Actions do not use the _latest_ PAC CLI -- they use an explicit version of PAC CLI hard coded for each version of PPBT and GH Actions. This is due to having no programably enforceable contract between the command line args of the C# based PAC CLI and the Task args sent to it from the NodeJS based ADO / GH Actions, so we do not update the version of PAC CLI used unless we know of bug fixes or new features that we need to ship for the pipeline tools.
> 
> We can look at updating to a new PAC version here in the next release cycle (probably next week?) to get your fix in.

Makes sense - thanks for the education and for looking at getting the fix into the next release cycle.

**Comment by tehcrashxor** [comment]
PPBT v2.0.18 has been released with an updated PAC v1.21.13 via PR #297, so @tomjschneider's fix should be deployed.
@concoursdude, please check your pipeline and re-open if the issue reoccurs.
